servicetalk-serializer-utils/src/main/java/io/servicetalk/serializer/utils/StringUtf8Serializer.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright © 2021 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.serializer.utils;
+
+import io.servicetalk.buffer.api.Buffer;
+import io.servicetalk.buffer.api.BufferAllocator;
+import io.servicetalk.serializer.api.SerializerDeserializer;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+/**
+ * Serialize/deserialize {@link String}s with {@link java.nio.charset.StandardCharsets#US_ASCII} encoding.
+ */
+public final class StringUtf8Serializer implements SerializerDeserializer<String> {
```

Comment:
I don't see any advantage over extending `StringCharsetSerializer`

Reply:
there are specialized methods on `Buffer` for `writeUtf8` and `writeUsAscii` but good point I can internalize that from the API perspective.